x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/utils.ts

```diff
@@ -53,3 +55,29 @@ export function crawlerDataServerToClient(payload: CrawlerDataFromServer): Crawl
     domains: domains.map((domain) => crawlerDomainServerToClient(domain)),
   };
 }
+
+export function crawlDomainValidationToResult(
+  data: CrawlerDomainValidationResultFromServer
+): CrawlerDomainValidationStep {
+  if (!data.valid) {
+    return {
+      state: 'invalid',
+      blockingFailure: true,
+      message: data.results.find((result) => result.result === 'failure')?.comment,
+    };
+  }
+
+  const warningResult = data.results.find((result) => result.result === 'warning');
```

Comment:
[SUPER optional] is `result.result` weird to read at all? i wonder if we can make it a bit more natural via destructuring
```suggestion
      message: data.results.find(({ result }) => result === 'failure')?.comment,
    };
  }

  const warningResult = data.results.find(({ result }) => result === 'warning');
```

hmmm... but then that makes the `?.comment` line a little more obtuse/harder to grok. ugh feel free to just ignore this, i'll collapse it

Reply:
I don't like the way it is now but I think I like the destructured version even less 😓